Core/src/IfcGeometryConverter/CurveConverter.h

```diff
@@ -2804,14 +2804,11 @@ namespace OpenInfraPlatform
 					double xRadius = ellipse->SemiAxis1 * this->UnitConvert()->getLengthInMeterFactor();
 					double yRadius = ellipse->SemiAxis2 * this->UnitConvert()->getLengthInMeterFactor();
 					
-					carve::math::Matrix inversePlacement = GeomUtils::computeInverse(placement);
-					carve::geom::vector<3> directionFromCenterToPoint = inversePlacement * point;
-					//https://www.algebra.com/algebra/homework/Quadratic-relations-and-conic-sections/Tangent-lines-to-an-ellipse.lesson
-					// if the radial vector from Center to a point has coordinates (a,b), then the direction vector at that point is (−b/yRadius^2 ,a/xRadius^2)
-					carve::geom::vector<3> direction = carve::geom::VECTOR(-directionFromCenterToPoint.y / (yRadius * yRadius), 
-																			directionFromCenterToPoint.x / (xRadius * xRadius), 
-																			0.);
-					return direction.normalize();
+					double angleFromX = std::acos(point.x / xRadius);
+					double angleFromY = std::asin(point.y / yRadius);
+					double angle = (angleFromX + angleFromY) * 0.5;

```

Comment:
I don't think this is correct. Why not arctan(point.y, point.x)?

Reply:
Yes it will be better approach, thank You!

Reply:
Resolved: 93d59f9